geometry/rp2_element.hpp

```diff
@@ -0,0 +1,65 @@
+﻿
+#pragma once
+
+#include <string>
+
+#include "geometry/r3_element.hpp"
+
+namespace principia {
+namespace geometry {
+namespace internal_rp2_element {
+
+// An |RP2Element| is an element of ℝP², the real projective plane.  |Scalar|
+// must be a 1-dimensional vector space over ℝ, typically represented by
+// |Quantity| or |double|.
+template<typename Scalar>
+class RP2Element {
+ public:
+  // Constructs a point from a set of homogeneous coordinates.  The
+  // coordinates may not all be zero.
+  explicit RP2Element(Scalar const& x, Scalar const& y, double z);
```

Comment:
Document the semantics: this class handles underflows gracefully, and behaves as if taking the limit z->0 before the limit of x and y.